packages/components/autocomplete/src/Autocomplete.tsx

```diff
@@ -0,0 +1,246 @@
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface AutocompleteProps<ItemType = any>
+  extends CommonProps,
+    Pick<
+      TextInputProps,
+      'isDisabled' | 'isInvalid' | 'isReadOnly' | 'isRequired' | 'id'
+    > {
+  /**
+   * It’s an array of data to be used as "options" by the autocomplete component.
+   */
+  items: ItemType[];
+  /**
+   * This is the function that will tell the component how the `items` should be filtered when the input value changes.
+   * It will be used in the `filter()` method of the array passed in the `items` prop and it needs two arguments: an item and the inputValue.
+   */
+  onFilter: (item: ItemType, inputValue: string) => void;
+  /**
+   * This is the function that will be called when the user selects one of the "options" in the list.
+   * It receives the selected item as an argument and it needs to return a string that will be set as the value of `TextInput`.
+   */
+  onSelectItem: (item: ItemType) => void;
+  /**
+   * This is the function that will be called for each "item" passed in the `items` prop.
+   * It receives the "item" as an argument and returns a ReactNode.
+   */
+  renderItem?: (item: ItemType) => React.ReactElement;
+  /**
+   * When using objects as `items`, we recommend passing a function that tells Downshift how to extract a string
+   * from those objetcs to be used as inputValue
+   */
+  itemToString?: (item: ItemType) => string;
+  /**
+   * If this is set to `true` the text input will be cleared after an item is selected
+   * @default false
+   */
+  clearAfterSelect?: boolean;
+  /**
+   * This is the value will be passed to the `placeholder` prop of the input.
+   * @default "Search"
+   */
+  placeholder?: string;
+  /**
+   * A message that will be shown when it is not possible to find any option that matches the input value
+   * @default "No matches"
+   */
+  noMatchesMessage?: string;
+  /**
+   * Use this prop to get a ref to the input element of the component
+   */
+  inputRef?: React.Ref<HTMLInputElement>;
+  /**
+   * Use this prop to get a ref to the toggle button of the component
+   */
+  toggleRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * Use this prop to get a ref to the list of items of the component
+   */
+  listRef?: React.Ref<HTMLUListElement>;
+}
+
+function _Autocomplete<ItemType>(
+  props: AutocompleteProps<ItemType>,
+  ref: React.Ref<HTMLDivElement>,
+) {
+  const {
+    id,
+    className,
+    clearAfterSelect = false,
+    items,
+    onFilter,
+    onSelectItem,
+    renderItem,
+    itemToString = (item: ItemType) => (item as unknown) as string,
+    isInvalid,
+    isDisabled,
+    isRequired,
+    isReadOnly,
+    noMatchesMessage = 'No Matches',
+    placeholder = 'Search',
+    inputRef,
+    toggleRef,
+    listRef,
+    testId = 'cf-autocomplete',
+  } = props;
+
+  const formProps = useFormControl({
+    id,
+    isInvalid,
+    isDisabled,
+    isRequired,
+    isReadOnly,
+  });
```

Comment:
You probably don't need `useFormControl` here. It will be too lengthy to explain it here, so I've created a PR
https://github.com/contentful/forma-36/pull/1342

Reply:
made a comment in the PR

I guess we need some documentation around this hook, I still don't get it